packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -279,45 +276,76 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     }
   }
 
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    super.addAllowedPointer(event);
+    if (_state == _DragState.ready) {
+      _initialButtons = event.buttons;
+    }
+    _addPointer(event);
+  }
+
+  @override
+  void addAllowedPointerPanZoom(PointerPanZoomStartEvent event) {
+    super.addAllowedPointerPanZoom(event);
+    startTrackingPointer(event.pointer, event.transform);
+    if (_state == _DragState.ready) {
+      _initialButtons = kPrimaryButton;
+    }
+    _addPointer(event);
+  }
+
   @override
   void handleEvent(PointerEvent event) {
     assert(_state != _DragState.ready);
-    if (!event.synthesized
-        && (event is PointerDownEvent || event is PointerMoveEvent)) {
+    if (!event.synthesized &&
+        (event is PointerDownEvent ||
+         event is PointerMoveEvent ||
+         event is PointerPanZoomStartEvent ||
+         event is PointerPanZoomUpdateEvent)) {
       final VelocityTracker tracker = _velocityTrackers[event.pointer]!;
       assert(tracker != null);
-      tracker.addPosition(event.timeStamp, event.localPosition);
-    }
-
-    if (event is PointerMoveEvent) {
-      if (event.buttons != _initialButtons) {
-        _giveUpPointer(event.pointer);
-        return;
+      if (event is PointerPanZoomStartEvent) {
+        tracker.addPosition(event.timeStamp, Offset.zero);
+      } else if (event is PointerPanZoomUpdateEvent) {
+        tracker.addPosition(event.timeStamp, event.pan);
+      } else {
+        tracker.addPosition(event.timeStamp, event.localPosition);
       }
+    }
+    if (event is PointerMoveEvent && event.buttons != _initialButtons) {
+      _giveUpPointer(event.pointer);
+      return;
+    }
+    if (event is PointerMoveEvent || event is PointerPanZoomUpdateEvent) {
+      final Offset delta = (event is PointerMoveEvent) ? event.delta : (event as PointerPanZoomUpdateEvent).panDelta;
+      final Offset localDelta = (event is PointerMoveEvent) ? event.localDelta : (event as PointerPanZoomUpdateEvent).localPanDelta;
+      final Offset position = (event is PointerMoveEvent) ? event.position : (event.position + (event as PointerPanZoomUpdateEvent).pan);
+      final Offset localPosition = (event is PointerMoveEvent) ? event.localPosition : (event.localPosition + (event as PointerPanZoomUpdateEvent).localPan);
```

Comment:
Nit: The very top of Flutter's coding style guide says [avoid using `is`](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#tldr).  Maybe there's no better way to do this without `is` but it jumped out at me so I wanted to call it out.  I realize that `handleEvent` was written this way before this PR too...

Reply:
I searched the style guide and only found the the following as details:

> ### Avoid interleaving multiple concepts together
> 
> Each API should be self-contained and should not know about other features.
Interleaving concepts leads to _complexity_.
>
> For example:
> 
> - Many Widgets take a `child`. Widgets should be entirely agnostic about the type
of that child. Don't use `is` or similar checks to act differently based on the
type of the child.
> ...

I think it doesn't apply to our situation.

Reply:
Yeah I'm sure this is fine for now.  The part of the style guide that I was referencing was in the tl;dr at the top:

 > Avoid `is`, `print`, `part of`, `extension` and `_`.

But yeah I don't see much in terms of *why* that is.